### ISSUE TITLE
Amorphous simulation with FrozenPhonon to do RDF analysis

### ISSUE THREAD

**Issue by Loewe313** [abTEM/abTEM#144]
Hi Jacob and Toma,

Firstly, thanks for your great job to create abTEM module.
I am trying to use a .cif file to do 4dstem simulation for subsequent RDF analysis. When analysing amorphous material, it is vital to calculate the background fit estimates which related to:
(1) thermal diffuse background, resulting from thermal motion of the atoms and which dominates the behavior shown here at low k values, 
(2) the single atom scattering factors, describing the scattered intensity profiles which result from individual atoms and which dominate the behavior at high k values. 

I have used FrozenPhonon to calculate the potential of the material to simulate vibration around their equilibrium position due to thermal and zero-point vibrations. I am totally confirmed that the .cif file is amorphous and the RDF plot part is right. Here are the results comparing the right situation with the wrong one.

Figure 1 the right figure from Microscopy and Microanalysis (2021), 27, 712–743; 
<img width="597" alt="1" src="https://github.com/abTEM/abTEM/assets/154033559/4902d5d8-c623-4a69-b57c-f9655faff9a2">
Figure 2 the wrong figure.
<img width="531" alt="2" src="https://github.com/abTEM/abTEM/assets/154033559/035b47ed-06e7-414e-af99-18f58548212a">

The order of magnitude and the trend of the curve are both wrong. I will be grateful for your kindly wise counsel. 

Thank you.

**Comment by TomaSusi** [comment]
Certainly something is not matching, but it's not possible to say what the issue could be without more information. Could you please share the structure and your code?

**Comment by Loewe313** [comment]
Thanks for your quick response.
Here is the structue and .ipynb code. Please unzip the package below.
[4dstem_frozen_phonon.zip](https://github.com/abTEM/abTEM/files/14489556/4dstem_frozen_phonon.zip)

**Comment by TomaSusi** [comment]
Thanks!

I see that ase.io.read is throwing a warning when you read in your cif file, I guess you verified that the structure is correct regardless?
> `crystal system 'tetragonal' is not interpreted for space group Spacegroup(1, setting=1). This may result in wrong setting!`

Second, your scan is quite odd: the syntax 
```
start=(11/30, 11/30),
end=(23/60, 23/60),
sampling=0.5
```
gives you just a tiny 5x5 grid at a single point essentially, is that really what you want to simulate?

If you wanted to limit the number of scan points, but sample the structure more randomly, I would rather recommend using
```
start=(0,0),
end=(1,1),
gpts=(5,5)
---

to scan the same number of points but over the entire cell (which is what I did below).

-------------------

After that, you run some py4DSTEM analyses, which I am not fully familiar with, but I'll try.

- I would only use this to create the datacube: `datacube = py4DSTEM.DataCube(data=measurement_4d.array)`, I'm not sure what your additional `__array__()` syntax was for?
- Do you really need a beamstop? If you block the center, you certainly cannot expect similar results for scattering vectors tending to zero, and the data you compare to did not use a beamstop.
- The sharp cutoff you see at higher scattering vectors is probably due to how you set up the PolarDatacube; could you try increasing qmax?

Unfortunately I cannot run the remaining analyses with my version of py4DSTEM, as I get a "NotImplementedError: The da.median function only works along an axis.  The full algorithm is difficult to do in parallel" when trying to fit the amorphous ring.

However, if you remove the beamstop, increase qmax, and use a linear y-axis instead of logarithmic, you might get results that are closer to what you expect. I would also advice to use a similar plotting style: if I understand correctly, the black curve in your plot corresponds to the red curve in the literature plot, and your red background curve should be colored something else.

---------------------

However, all this being said, the comparison you are trying to make is never going be that similar for a number of reasons:

- The literature data you are comparing to is for amorphous silicon, so bond lengths will be different.
- That data is experimental so it will be to some degree decoherent, but most importantly, significant diffuse backgrounds due to inelastic scattering are not being modeled.
- Your model is quite thin (only 10 A!), so most of your scattered intensity will be in the forward direction, very different to the presumably thicker experimental sample 
 
This last point is the main reason for the very different intensities, and you probably need to increase the thickness of your model significantly, or simply normalize the intensity. You see visualize this easily by plotting the azimuthal average for the scan-point summed diffraction pattern with abTEM:

```
measurement_4d.sum((0,1)).azimuthal_average(max_angle=100).show(
    overlay=True, legend=True, units="mrad"
)
plt.yscale('log')
```

![image](https://github.com/abTEM/abTEM/assets/66075496/9f2e2ed1-e71d-48e4-85fd-bdd405249fd3)

**Comment by Loewe313** [comment]
Thanks Toma!

So kind of you to response so quickly. And I took some time to read the comment carefully and really learned a lot.

Firstly, the warning is definitely worth thinking. In my research, I think tetragonal structures can be treated as orthogonal structures.  

Secondly, the selected scan area is a temporary try to test somewhere with medium-range order in amorphous. Small selected areas can be calculated more quickly. Maybe there is some potential risk. And to scan the whole area is supposed at last.

Thirdly, some detailed code in py4dstem part contributes to mixing abtem with py4dstem. I am not familiar with zarr. It wastes lots of time to compute() every time when changing the parameters above. But the advice is quite useful for me to make it more brief and accurate.

At the beginning, I tried to get a familiar pdf plot with the literature data in order of magnitudes, not totally same in profile.  Afterwards, I figure out that it is not that necessary to get the same order of magnitude at Y-axis as what experimental data can generate. What matters are the distances between the peaks and the original at every pdf plot from all diffraction patterns. Those plots bring lots of information about amorphous. I think the results at present show some positive signals at least.

Besides, inelastic plasmon scattering and a more thick model are quite important as well. I will solve the problems.

Thanks again for being so patient and careful to show how to get onto the right path.

Best wishes.
